simulator-ui/tests/scenario.spec.ts

```diff
@@ -0,0 +1,219 @@
+  await page.getByTestId('itemsPerPageSelect').selectOption('10');
+
+})
+
+test('should display all scenario information of a starter scenario', async ({page}) => {
+  await mockBackendResponse(page, '**/api/scenarios**', {"name": "Test", "type": "STARTER"} );
+
+
+  await page.goto('http://localhost:9000/scenario');
+
+  await expect(page.getByTestId('scenarioEntitiesTable')).toBeVisible();
+  await expect(page.getByTestId('scenarioEntitiesName')).toHaveText('Test');
+  await expect(page.getByTestId('scenarioEntitiesType')).toHaveText('STARTER');
+  await expect(page.getByTestId('scenarioLaunchButton')).toBeVisible();
+  await expect(page.getByTestId('scenarioExecutionButton')).toBeVisible();
+});
+
+test('should display all scenario information of a non-starter scenario', async ({page}) => {
+  await page.route('**/api/scenarios**', async route => {
+    await route.fulfill({json: [{"name": "Test", "type": "MESSAGE_TRIGGERED"}]});
+  });
+
+  await page.goto('http://localhost:9000/scenario');
+
+  await expect(page.getByTestId('scenarioEntitiesName')).toHaveText('Test');
+  await expect(page.getByTestId('scenarioEntitiesType')).toHaveText('MESSAGE_TRIGGERED');
+  await expect(page.getByTestId('scenarioLaunchButton')).toBeHidden();
+  await expect(page.getByTestId('scenarioExecutionButton')).toBeVisible();
+});
+
+test('should have the first 10 of 12 elements displayed in the table and the pagination field should display it right', async ({page}) => {
+  await checkIfAllJsonContentIsVisible(page, 10, scenarioJson);
+});
+
+// should one test EVERY option?
+test('should have all 12 scenarios displayed in the table after selecting 20 as table size', async ({page}) => {
+  const selectedPageSize = '20';
+  await page.getByTestId('itemsPerPageSelect').selectOption(selectedPageSize);
+
+  await expect(page.getByTestId('itemsPerPageSelect')).toHaveValue(selectedPageSize);
+  for (const element of scenarioJson) {
+    await expect(page.getByText(element.name)).toBeVisible();
+  }
+  await expect(page.getByText('Showing 1-12 of 12 Items')).toBeVisible();
+});
+
+// how do you write this kind of test correctly? should the filter test be in the same as the reset button? or separate? Is the independence of the tests violated?
+test('text filter input should apply and clear filter button should reset it', async ({page}) => {
+  await applyFilterAAndCheckCorrectness(page);
+  await page.getByTestId('clearFilterButton').click();
+
+  await expect(page.getByTestId('scenarioFilterByNameInput')).toHaveValue('a');
+  await checkIfAllJsonContentIsVisible(page, 10, scenarioJson);
+});
+
+// is the test independence again violated? should there be two separate tests for the pagination-select-option-button and the refresh button? --> no because the refresh button affects the pagination number inevitably.
+test('should have updated displayed scenarios after refresh button was clicked and a new scenario received', async ({page}) => {
```

Comment:
Is the test independence violated again? should there be two separate tests for the pagination-select-option-button and the refresh button? --> no because the refresh button affects the pagination number inevitably.

Reply:
ok